### PR TITLE
#205 taking values from identification extension when missing from core

### PR DESCRIPTION
**Pull request by marcos-lg**
#205 Basically, if it's an identification field and all the identification fields in the core are null we take the values of these fields from the Identification extension if exists. If there are multiple identification extensions we take it from the most recent one but only if we can determine which one it's the most recent.

### CODE REVIEW

File: sdks/core/src/main/java/org/gbif/pipelines/core/parsers/temporal/TemporalParser.java [hunk 1]
```diff
@@ -209,4 +209,34 @@ protected static boolean isValidDate(
...
     return likelyRange.contains(LocalDate.of(year, month, day));
   }
+
+  public static Optional<Long> getTime(TemporalAccessor temporalAccessor) {
+    if (temporalAccessor == null) {
+      return Optional.empty();
+    }
+
+    int year;
+    if (temporalAccessor.isSupported(ChronoField.YEAR)) {
```
Comment (timrobertson100) [inline sdks/core/src/main/java/org/gbif/pipelines/core/parsers/temporal/TemporalParser.java:219]: Are these changes intended to be in this commit?

**Comment by timrobertson100** [comment]
The logic in here looks good to me. 
I think a few methods intended for chrono work have slipped into the commit by mistake which might be worth checking.

File: sdks/core/src/main/java/org/gbif/pipelines/core/parsers/temporal/TemporalParser.java (hunk 1 above)
Comment (timrobertson100) [reply sdks/core/src/main/java/org/gbif/pipelines/core/parsers/temporal/TemporalParser.java:219]: OK - it was just the reference to `ChronoField` that jumped out as chrono isn't used in the [identification history extension](https://rs.gbif.org/extension/dwc/identification_history_2022-02-02.xml). Made me just wonder if that was unrelated work slipping in

Comment (marcos-lg) [reply sdks/core/src/main/java/org/gbif/pipelines/core/parsers/temporal/TemporalParser.java:219]: Ah it's because I use the `dateIdentified ` field of the extension for the sorting and since it's a string I have to parse it and convert it into a date.

File: sdks/core/src/main/java/org/gbif/pipelines/core/utils/ModelUtils.java [hunk 2]
```diff
@@ -106,4 +148,70 @@ public static void addIssue(Issues model, Set<String> issues) {
   public static void addIssueSet(Issues model, Set<OccurrenceIssue> issues) {
     issues.forEach(x -> addIssue(model, x));
   }
+
+  public static Map<String, String> getIdentificationFieldTermsSource(ExtendedRecord er) {
```
[the remark this answers is not in the document]
Comment (marcos-lg) [reply sdks/core/src/main/java/org/gbif/pipelines/core/utils/ModelUtils.java:152]: ok, I'll move it to a class in the same package and I'll also move the function in the temporalParser

[the remark this answers is not in the document]
Comment (marcos-lg) [reply sdks/core/src/main/java/org/gbif/pipelines/core/utils/ModelUtils.java]: just pushed it

**Review by muttcg — APPROVED** [review]
